src/test/ui/rfc-2632-const-trait-impl/const-impl-requires-const-trait.rs

```diff
@@ -5,5 +5,6 @@ pub trait A {}
 
 impl const A for () {}
 //~^ ERROR: const `impl`s must be for traits marked with `#[const_trait]`
+//~| ERROR: ~const can only be applied to `#[const_trait]` traits
```

Comment:
This emission site is a bit buggy, Not sure yet how to best resolve it. Maybe we could remove the other check and figure out how to tune the message at the site of this new error